cmake/helpers/CheckDependentLibraries.cmake

```diff
@@ -173,114 +186,79 @@ endif ()
 
 find_package(PROJ 6.0 REQUIRED)
 
-find_package(TIFF 4.0)
-if (TIFF_FOUND)
-  set(HAVE_TIFF ON)
-  set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${TIFF_INCLUDE_DIR})
-  set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${TIFF_LIBRARIES})
-  set(GDAL_USE_LIBTIFF_INTERNAL
-      OFF
-      CACHE BOOL "")
-endif ()
-if (NOT HAVE_TIFF)
-  set(GDAL_USE_LIBTIFF_INTERNAL
-      ON
-      CACHE BOOL "")
-endif ()
+gdal_check_package(TIFF "Support for the Tag Image File Format (TIFF)." VERSION 4.0 CAN_DISABLE)
 set_package_properties(
   TIFF PROPERTIES
   URL "https://libtiff.gitlab.io/libtiff/"
-  DESCRIPTION "support for the Tag Image File Format (TIFF)."
+  DESCRIPTION "Support for the Tag Image File Format (TIFF)."
   TYPE RECOMMENDED)
+invert_on_off(GDAL_USE_TIFF NOT_GDAL_USE_TIFF)
+set(GDAL_USE_LIBTIFF_INTERNAL
+    ${NOT_GDAL_USE_TIFF}
+    CACHE BOOL "Use internal libtiff copy (if set to ON, has precedence over GDAL_USE_TIFF)")
+if ((NOT GDAL_USE_TIFF) AND (NOT GDAL_USE_LIBTIFF_INTERNAL))
+  message(FATAL_ERROR "GDAL_USE_TIFF or GDAL_USE_LIBTIFF_INTERNAL must be set to ON")
+endif ()
 
 gdal_check_package(ZSTD "ZSTD compression library" CAN_DISABLE)
 gdal_check_package(SFCGAL "gdal core supports ISO 19107:2013 and OGC Simple Features Access 1.2 for 3D operations"
                    CAN_DISABLE)
 
 gdal_check_package(GeoTIFF "libgeotiff library (external)" CAN_DISABLE)
-if (NOT GDAL_USE_GEOTIFF)
-  set(GDAL_USE_LIBGEOTIFF_INTERNAL
-      ON
-      CACHE BOOL "Use internal libgeotiff copy (if set to ON, has precedence over GDAL_USE_GEOTIFF)")
-  if (NOT GDAL_USE_LIBGEOTIFF_INTERNAL)
-    message(FATAL_ERROR "GDAL_USE_GEOTIFF or GDAL_USE_LIBGEOTIFF_INTERNAL must be set to ON")
-  endif ()
-else ()
-  set(GDAL_USE_LIBGEOTIFF_INTERNAL
-      OFF
-      CACHE BOOL "Use internal libgeotiff copy (if set to ON, has precedence over GDAL_USE_GEOTIFF)")
+invert_on_off(GDAL_USE_GEOTIFF NOT_GDAL_USE_GEOTIFF)
+set(GDAL_USE_LIBGEOTIFF_INTERNAL
```

Comment:
I would prefer if the variable was named `GDAL_USE_GEOTIFF_INTERNAL`, similar to `GDAL_USE_GEOTIFF`.

Reply:
done